simulation/simulation_endpoint.py

```diff
@@ -11,3 +13,16 @@ class SimulationEndpoint(AutoMockEndpoint):
     async def open(self):
         self._open = True
         return succeed(None)
+
+    def get_link_latency(self, to_address):
+        """
+        Return the link latency to a particular destination address, in seconds.
+        """
+        return 0
+
+    async def actual_send(self, socket_address, packet):
+        await sleep(self.get_link_latency(socket_address))
+        super().send(socket_address, packet)
+
+    def send(self, socket_address, packet):
+        ensure_future(self.actual_send(socket_address, packet))
```

Comment:
It seems like `actual_send` will not be overwritten in subclasses. If that is the case, please use this instead:

```suggestion
    def send(self, socket_address, packet):
        get_event_loop().call_later(self.get_link_latency(socket_address), super().send,
                                    socket_address, packet)
```

Reply:
That's a more clean solution indeed, thanks for the suggestion! 👍